### PR TITLE
100039_login_logout (step 1: apply gem devise)

### PR DESCRIPTION
**Pull request by TuBaDuc**

### CODE REVIEW

File: config/environments/development.rb [hunk 1]
```diff
@@ -51,4 +51,7 @@
   # Use an evented file watcher to asynchronously detect changes in source code,
   # routes, locales, etc. This feature depends on the listen gem.
   config.file_watcher = ActiveSupport::EventedFileUpdateChecker
+
+  # P2: gem devise
+  config.action_mailer.default_url_options = { host: "localhost", port: 3000 }
```
Comment (dieunb) [inline config/environments/development.rb]:
```
config.action_mailer.default_url_options = {host: "localhost", port: 3000}
```

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class User < ApplicationRecord
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable
```
Comment (hungnh103) [inline app/models/user.rb]: indent

File: config/routes.rb [hunk 3]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
+  devise_for :users
   root "static_pages#home"
   get "/help",  to: "static_pages#help"
```
Comment (hungnh103) [inline config/routes.rb]: remove whitespace

File: config/routes.rb [hunk 4]
```diff
@@ -1,4 +1,5 @@
...
   get "/about",  to: "static_pages#about"
```
Comment (hungnh103) [inline config/routes.rb]: as above mentioned

**Pull request merged**